src/helm/benchmark/annotation/annotator.py

```diff
@@ -0,0 +1,45 @@
+from typing import Dict, Any, List
+from abc import abstractmethod, ABC
+from dataclasses import dataclass
+
+from helm.benchmark.adaptation.request_state import RequestState
+from helm.common.object_spec import ObjectSpec
+
+
+class Annotator(ABC):
+    """Annotator is an abstract class for annotating a request state. Annotators are used to add additional
+    information to a request state that is needed for a metric to understand the request. This could be
+    parsing, rendering an image based on the text completion, etc."""
+
+    def __init__(self, name: str):
+        self.name = name
+
+    @abstractmethod
+    def annotate(self, request_state: RequestState) -> List[Dict[str, Any]]:
+        """Fills the annotations field of the request state with additional information
+        that are implementation specific."""
+        pass
+
+
+@dataclass(frozen=True)
+class AnnotatorSpec(ObjectSpec):
+    """Specifies how to create an `Annotator`."""
+
+    name: str = "annotator"  # Must be initialized since ObjectSpec has default value for args
```

Comment:
Prefer not to have this as a field in the spec - options are:

- Make `name` an `arg`
- Cache just using the class name (and disallow args)
- Cache using tuple of class name and a canonical flattening of args

Reply:
I chose the second option and added a comment to specify that people should not use `args`. I had to keep `args` to use the `inject` function.